### PR TITLE
Feat support alternative gas token

### PR DESCRIPTION
**Pull request by yiweichi**
### Purpose or design rationale of this PR

1. to support use alternative gas token
2. when using alternative gas token instead of Ether, check gas token balance of gasTokenGateway contract instead of Ether balance of scrollMessenger contracts

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

### CODE REVIEW

File: conf/config.json [hunk 1]
```diff
@@ -13,11 +13,13 @@
...
         "usdc_gateway": "0x0000000000000000000000000000000000000000",
-        "lido_gateway": "0x0000000000000000000000000000000000000000"
+        "lido_gateway": "0x0000000000000000000000000000000000000000",
+        "gas_token_gateway": "0x8587FE761F1669Fa75C444c8BC959c28a64AD822"
       },
       "scroll_messenger": "0x50c7d3e7f7c656493D1D76aaa1a836CedfCBB16A",
       "message_queue": "0xF0B2293F5D834eAe920c6974D50957A1732de763",
-      "scroll_chain": "0x2D567EcE699Eabe5afCd141eDB7A4f2D0D6ce8a0"
+      "scroll_chain": "0x2D567EcE699Eabe5afCd141eDB7A4f2D0D6ce8a0",
+      "gas_token": "0x0000000000000000000000000000000000000000"
```
Comment (colinlyguo) [inline conf/config.json:22]: is this a placeholder?

File: conf/config.json [hunk 2]
```diff
@@ -13,11 +13,13 @@
         "erc1155_gateway": "0xa5Df8530766A85936EE3E139dECE3bF081c83146",
         "dai_gateway": "0x0000000000000000000000000000000000000000",
         "usdc_gateway": "0x0000000000000000000000000000000000000000",
-        "lido_gateway": "0x0000000000000000000000000000000000000000"
+        "lido_gateway": "0x0000000000000000000000000000000000000000",
```
Comment (colinlyguo) [inline conf/config.json:16]: Should we add the wrapped token gateway as well to track eth deposit?

Comment (colinlyguo) [reply conf/config.json:16]: or does it use the following gateway?
```
    /// @notice The address of ERC20 gateway used to bridge WETH.
    address public immutable gateway;
```

Comment (yiweichi) [reply conf/config.json:16]: It use L1StandardERC20Gateway.

Comment (colinlyguo) [reply conf/config.json:16]: got it.

File: conf/config.json (hunk 1 above)
Comment (colinlyguo) [reply conf/config.json:22]: I see. when it's a zero address, it's eth, when it's a non-zero address, it's the gas token (to get decimals).

**Review by colinlyguo — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Review by colinlyguo — APPROVED** [review]

**Pull request merged**